src/main/java/gyro/google/compute/InstanceAttachedDiskInitializeParams.java

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright 2019, Perfect Sense, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gyro.google.compute;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.google.api.services.compute.model.AttachedDiskInitializeParams;
+import gyro.core.resource.Diffable;
+import gyro.core.validation.Regex;
+import gyro.google.Copyable;
+
+public class InstanceAttachedDiskInitializeParams extends Diffable implements Copyable<AttachedDiskInitializeParams> {
+
+    private String sourceImage;
```

Comment:
is this required ?

Reply:
No, when creating a new instance, one of initializeParams.sourceImage or initializeParams.sourceSnapshot or disks.source is required. I haven't put the other two fields on there (yet).